File: ios/MRT/Core/Models/SessionModel.swift
import Foundation

struct SessionModel: Identifiable, Equatable {
    let id: String
    let name: String
    let status: Mrt_TaskStatus
    let createdAtMs: UInt64
    let lastActiveMs: UInt64
    let workingDirectory: String

    init(
        id: String,
        name: String,
        status: Mrt_TaskStatus,
        createdAtMs: UInt64,
        lastActiveMs: UInt64,
        workingDirectory: String
    ) {
        self.id = id
        self.name = name
        self.status = status
        self.createdAtMs = createdAtMs
        self.lastActiveMs = lastActiveMs
        self.workingDirectory = workingDirectory
    }

    init(_ session: Mrt_SessionInfo) {
        self.init(
            id: session.sessionID,
            name: session.name,
            status: session.status,
            createdAtMs: session.createdAtMs,
            lastActiveMs: session.lastActiveMs,
            workingDirectory: session.workingDir
        )
    }
}

File: android/app/src/main/java/com/mrt/app/core/models/SessionModel.kt
package com.mrt.app.core.models

import mrt.Mrt

data class SessionModel(
    val id: String,
    val name: String,
    val status: Mrt.TaskStatus,
    val createdAtMs: Long,
    val lastActiveMs: Long,
    val workingDirectory: String,
) {
    companion object {
        fun fromProto(session: Mrt.SessionInfo): SessionModel =
            SessionModel(
                id = session.sessionId,
                name = session.name,
                status = session.status,
                createdAtMs = session.createdAtMs,
                lastActiveMs = session.lastActiveMs,
                workingDirectory = session.workingDir,
            )
    }
}
